### DISCUSSION TITLE
Is there limitation to "project metadata" for single-file Haskell script?

### DISCUSSION THREAD

**Discussion by hellwolf** [haskell/cabal#9084] [General]
I would like to configure "vendor supplied dependencies" in a single-file Haskell script run via `cabal run`, as the documentation [suggested using "project metadata"](https://cabal.readthedocs.io/en/stable/getting-started.html#run-a-single-file-haskell-script):

```
{- project:
packages: pkgs/loliyul, pkgs/examples
-}
{- cabal:
build-depends: loliyul, loliyul-examples
-}

import Basic

main = pure ()
```

But it seems to be "packages" meta data is not supported in the project metadata.

Is there any limitation to this "project metadata" which is not very evident from the documentation?

**Comment by andreabedini** [comment]
Unfortunately there are indeed serious limitations. I refer you to #8024 for more details.

**Comment by ulysses4ever** [comment]
This has been answered above.
